File: tree/easy/is_same_tree_100.go
package easy

import "github.com/vielendanke/algorithms_in_place/tree"

func isSameTree(p *tree.TreeNode, q *tree.TreeNode) bool {
	if p == nil && q == nil {
		return true
	}
	if p == nil || q == nil {
		return false
	}
	if p.Val != q.Val {
		return false
	}
	return isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
}

File: leetcode/tree/easy/java_solutions/IsSameTree_100.java
package leetcode.tree.easy.java_solutions;

import static leetcode.tree.Helper.TreeNode;

public class IsSameTree_100 {

    private static class Solution {
        public boolean isSameTree(TreeNode p, TreeNode q) {
            if (p == null && q == null) {
                return true;
            } else if (p == null || q == null) {
                return false;
            } else {
                if (p.val != q.val) {
                    return false;
                }
                return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
            }
        }
    }
}
